nightfall-client/src/utils/getCommitmentInfo.mjs

```diff
@@ -0,0 +1,139 @@
+    oldCommitments = await findUsableCommitmentsMutex(
+      compressedZkpPublicKey,
+      ercAddress,
+      tokenId,
+      generalise(transferValue),
+    );
+    if (oldCommitments) {
+      logger.debug(`Found commitments ${JSON.stringify(oldCommitments, null, 2)}`);
+    } else {
+      throw new Error('No suitable commitments were found'); // caller to handle - need to get the user to make some commitments or wait until they've been posted to the blockchain and Timber knows about them
+    }
+  } else {
+    throw new Error('No suitable commitments were found'); // caller to handle - need to get the user to make some commitments or wait until they've been posted to the blockchain and Timber knows about them
+  }
+  // Having found either 1 or 2 commitments, which are suitable inputs to the
+  // proof, the next step is to compute their nullifiers;
+  const nullifiers = oldCommitments.map(commitment => new Nullifier(commitment, nullifierKey));
+  // then the new output commitment(s)
+  const totalInputCommitmentFeeValue = oldCommitments.reduce(
+    (acc, commitment) => acc + commitment.preimage.value.bigInt,
+    0n,
+  );
+
+  // time for a quick sanity check.  We expect the number of old commitments and nullifiers to be equal.
+  if (nullifiers.length !== oldCommitments.length) {
+    logger.error(
+      `number of old commitments: ${oldCommitments.length}, number of nullifiers: ${nullifiers.length}`,
+    );
+    await Promise.all(oldCommitments.map(o => clearPending(o)));
+    throw new Error('Number of nullifiers and old commitments are mismatched');
+  }
+
+  // we may need to return change to the recipient
+  const change = totalInputCommitmentFeeValue - transferValue - addedFee;
+
+  // if so, add an output commitment to do that
+  if (change !== 0n) {
+    valuesArray.push(new GN(change));
+    recipientZkpPublicKeysArray.push(zkpPublicKey);
+  }
+
+  try {
+    const salts = await Promise.all(valuesArray.map(async () => randValueLT(BN128_GROUP_ORDER)));
+
+    // Generate new commitments, already truncated to u32[7]
+    const newCommitments = valuesArray.map(
+      (value, i) =>
+        new Commitment({
+          ercAddress,
+          tokenId,
+          value,
+          zkpPublicKey: recipientZkpPublicKeysArray[i],
+          salt: salts[i].bigInt,
+        }),
+    );
+
+    // Commitment Tree Information
+    const commitmentTreeInfo = await Promise.all(oldCommitments.map(c => getSiblingInfo(c)));
+    const localSiblingPaths = commitmentTreeInfo.map(l => {
+      const path = l.siblingPath.path.map(p => p.value);
+      return generalise([l.root].concat(path.reverse()));
+    });
+    const leafIndices = commitmentTreeInfo.map(l => l.leafIndex);
+    const blockNumberL2s = commitmentTreeInfo.map(l => l.isOnChain);
+    const roots = commitmentTreeInfo.map(l => l.root);
+    logger.info(
```

Comment:
Construction doesnt work. Change to -> 
    logger.info(
      `Constructing transfer transaction with blockNumberL2s ${blockNumberL2s} and roots ${roots}`);